aptos-move/framework/src/aptos.rs

```diff
@@ -3,89 +3,140 @@
 
 #![forbid(unsafe_code)]
 
-use move_deps::{
-    move_binary_format::file_format::CompiledModule,
-    move_compiler::{
-        compiled_unit::{CompiledUnit, NamedCompiledModule},
-        shared::NumericalAddress,
-    },
-    move_package::compilation::compiled_package::CompiledPackage,
-};
+use crate::release_builder::RELEASE_BUNDLE_EXTENSION;
+use crate::release_bundle::ReleaseBundle;
+use crate::{path_in_crate, BuildOptions, ReleaseOptions};
+use clap::ArgEnum;
 use once_cell::sync::Lazy;
-use std::collections::{BTreeMap, HashSet};
-use std::hash::Hash;
+use std::fmt::Display;
+use std::path::PathBuf;
+use std::str::FromStr;
 
-const APTOS_FRAMEWORK_DIR: &str = "aptos-framework/sources";
-const APTOS_STDLIB_DIR: &str = "aptos-stdlib/sources";
-const MOVE_STDLIB_DIR: &str = "move-stdlib/sources";
-const TOKEN_MODULES_DIR: &str = "aptos-token/sources";
-static APTOS_PKG: Lazy<CompiledPackage> = Lazy::new(|| super::package("aptos-framework"));
-static TOKEN_PKG: Lazy<CompiledPackage> = Lazy::new(|| super::package("aptos-token"));
-static APTOS_STDLIB_PKG: Lazy<CompiledPackage> = Lazy::new(|| super::package("aptos-stdlib"));
+// ===============================================================================================
+// Release Targets
 
-pub fn dedup<T, K: Hash + Eq, F: Fn(&T) -> K>(lists: Vec<T>, f: F) -> Vec<T> {
-    let mut res = vec![];
-    let mut keys = HashSet::new();
-    for l in lists {
-        let key: K = f(&l);
-        if keys.insert(key) {
-            res.push(l);
-        }
-    }
-    res
+/// Represents the available release targets. `Current` is in sync with the current client branch,
+/// which is ensured by tests.
+#[derive(ArgEnum, Clone, Copy, Debug)]
+pub enum ReleaseTarget {
+    Current,
+    Devnet,
+    Testnet,
+    Mainnet,
 }
 
-pub fn files() -> Vec<String> {
-    let mut files = super::move_files_in_path(MOVE_STDLIB_DIR);
-    files.extend(super::move_files_in_path(APTOS_STDLIB_DIR));
-    files.extend(super::move_files_in_path(APTOS_FRAMEWORK_DIR));
-    files.extend(super::move_files_in_path(TOKEN_MODULES_DIR));
-    files.extend(super::move_files_in_path(APTOS_STDLIB_DIR));
-    dedup(files, |f| f.to_string())
+impl Display for ReleaseTarget {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let str = match self {
+            ReleaseTarget::Current => "current",
```

Comment:
Maybe we should rename `current` to `development` or `release-candidate` to show that it's not actually deployed anywhere.